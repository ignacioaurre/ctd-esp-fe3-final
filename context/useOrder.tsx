import { OrderContext, OrderContextState } from ".";
import { useContext } from "react";

const useOrder = (): OrderContextState => {
    const context = useContext(OrderContext);
    if (!context) {
        throw new Error(
            'useOrder must be used within a OrderProvider'
        );
    }
    return context;
};

export default useOrder;