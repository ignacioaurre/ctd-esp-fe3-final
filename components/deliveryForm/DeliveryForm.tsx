import Stack from '@mui/material/Stack'
import React, { FC, useEffect } from 'react'
import ControlledTextInput from '../controlledTextInput/controlledTextInput'
import { FormProvider, useForm } from 'react-hook-form';
import *  as yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup'
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import useOrder from 'dh-marvel/context/useOrder';
import { Delivery } from 'dh-marvel/features/Types/state.types';
import { submitDelivery } from 'dh-marvel/context/actions';

export const deliverySchema = yup.object({
    direccion: yup.string().required("La dirección es requerida"),
    departamento: yup.string(),
    ciudad: yup.string().required("La ciudad es requerida"),
    provincia: yup.string().required("La provincia es requerida"),
    codigoPostal: yup.string().required("El codigo postal es requerido"),
})

type DeliveryFormProps = {
    title: string,
    activeStep: number,
    handleBack: () => void,
    handleNext: () => void,
    steps: string[]
}

const DeliveryForm: FC<DeliveryFormProps> = ({title, activeStep, handleBack, handleNext, steps}: DeliveryFormProps) => {

    const { state: {order: {delivery: { direccion, departamento, ciudad, provincia, codigoPostal}}}, dispatch } = useOrder();

    const methods = useForm<Delivery>({
        resolver: yupResolver(deliverySchema),
            defaultValues: {
                direccion,
                departamento,
                ciudad,
                provincia,
                codigoPostal,
            }
    })

    const { setFocus, handleSubmit } = methods;

    const onSubmit = (data: Delivery) => {
        submitDelivery(dispatch, data);
        handleNext();
    }

    useEffect(() => {
        setFocus("direccion")
    }, [setFocus])

  return (
    <FormProvider {...methods} >
        <form onSubmit={handleSubmit(onSubmit)}>
        <h4>{title}</h4>
        <Stack>
            <ControlledTextInput name="direccion" label="Dirección " defaultValue="" />
            <ControlledTextInput name="departamento" label="Departamento " defaultValue="" />
            <ControlledTextInput name="ciudad" label="Ciudad " defaultValue="" />
            <ControlledTextInput name="provincia" label="Provincia " defaultValue="" />
            <ControlledTextInput name="codigoPostal" label="Codigo Postal " defaultValue="" />
        </Stack>
        <Box sx={{ display: 'flex', flexDirection: 'row', pt: 2 }}>
            <Button
            color="inherit"
            disabled={activeStep === 0}
            onClick={handleBack}
            sx={{ mr: 1 }}
            >
            Anterior
            </Button>
            <Box sx={{ flex: '1 1 auto' }} />
            <Button type="submit">
            {activeStep === steps.length - 1 ? 'Finalizar' : 'Proximo'}
            </Button>
        </Box>
        </form>
    </FormProvider>
  )
}

export default DeliveryForm