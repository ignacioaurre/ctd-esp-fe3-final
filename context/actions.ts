import { Card, Delivery, Register, ComicInfo } from 'dh-marvel/features/Types/state.types';
import { Dispatch } from 'react';
import { Action } from './reducer';

export const selectComic = (dispatch: Dispatch<Action>, comic: ComicInfo) => {
    dispatch({type: 'ADD_COMIC', payload: comic})
}

export const submitRegister = (dispatch: Dispatch<Action>, register: Register) => {
    dispatch({type: 'SUBMIT_REGISTER', payload: register});
}

export const submitDelivery = (dispatch: Dispatch<Action>, delivery: Delivery) => {
    dispatch({type: 'SUBMIT_DELIVERY', payload: delivery});
}

export const submitCard = (dispatch: Dispatch<Action>, card: Card) => {
    dispatch({type: 'SUBMIT_CARD', payload: card});
}

export const submitForm = (dispatch: Dispatch<Action>) => {
    dispatch({ type: 'SUBMIT_FORM'})
};

export const restoreState = (dispatch: Dispatch<Action>) => {
    dispatch({ type: 'RESTORE_STATE' })
}

export const setSnackbar = (dispatch: Dispatch<Action>, snackbar: string) => {
    dispatch({ type: 'SET_SNACKBAR', payload: snackbar})
}