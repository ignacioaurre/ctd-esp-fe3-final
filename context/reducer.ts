import { OrderState } from "context";
import { Comic } from "dh-marvel/features/Types/comic.types"
import { Card, Delivery, Register, Order, ComicInfo } from "dh-marvel/features/Types/state.types"

export const types = {
    ADD_COMIC: 'ADD_COMIC',
    SUBMIT_REGISTER: 'SUBMIT_REGISTER',
    SUBMIT_DELIVERY: 'SUBMIT_DELIVERY',
    SUBMIT_CARD: 'SUBMIT_CARD', 
}

export type Action = 
  |  { type: 'ADD_COMIC' , payload: ComicInfo}
  |  { type: 'SUBMIT_REGISTER', payload: Register }
  |  { type: 'SUBMIT_DELIVERY', payload: Delivery }
  |  { type: 'SUBMIT_CARD', payload: Card };

export const initialState: OrderState = { 
    order: {
        comic: {
            img: "",
            price: 0,
            title: "",
        },
        register: {
            usuario: "",
            apellido: "",
            email: "",
        },
        delivery: {
            direccion: "Siempre Viva 123",
            departamento: "",
            ciudad: "",
            provincia: "",
            codigoPostal: "",
        },
        card: {
            nroTarjeta: "",
            nombreTarjeta: "",
            fechaExp: "",
            codSeguridad: "",
        }
    }
}

export const reducer = (state: OrderState, action: Action) => {
    switch (action.type) {
        case 'ADD_COMIC':
            return {...state, comic: action.payload}
        case 'SUBMIT_REGISTER':
            return {...state, register: action.payload}
        case 'SUBMIT_DELIVERY':
            return {...state, delivery: action.payload}
        case 'SUBMIT_CARD':
            return {...state, card: action.payload}
        default:
      return state;
    }
}