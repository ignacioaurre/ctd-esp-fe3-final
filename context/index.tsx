import { Order } from 'dh-marvel/features/Types/state.types';
import React, { useMemo, useReducer, createContext, FC, Dispatch, PropsWithChildren } from 'react';

import { reducer, initialState, Action } from './reducer';

export interface OrderState {
  order: Order
}

export type OrderContextState = {
  state: {order: Order},
  dispatch: Dispatch<Action>
}

export const OrderContext = createContext<OrderContextState | undefined>(undefined);

const ContextProvider: FC<PropsWithChildren> = ({ children }: PropsWithChildren ) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  const contextValue = useMemo(
    () => ({
      state,
      dispatch,
    }),
    [state, dispatch],
  );

  return (
        <OrderContext.Provider value={contextValue} >
          {children}
        </OrderContext.Provider>
      )

};

export default ContextProvider;