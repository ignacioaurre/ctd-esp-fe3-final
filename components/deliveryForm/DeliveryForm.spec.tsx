import {render, screen, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import useOrder from 'dh-marvel/context/useOrder';
import ContextProvider from '../../context';
import { initialState } from 'dh-marvel/context/reducer';
import DeliveryForm, { deliverySchema } from './DeliveryForm';
import { Delivery } from 'dh-marvel/features/Types/state.types';
import { validCard } from 'dh-marvel/pages/api/checkout.route';

jest.mock("../../context/useOrder")
const mockUseOrder = useOrder as jest.MockedFunction<typeof useOrder>
const mockDispatch = jest.fn();
mockUseOrder.mockReturnValue({
    state: initialState,
    dispatch: mockDispatch
})
const activeStep = 0;
const handleNext = jest.fn();
const handleBack = jest.fn();
const steps = ['Datos personales', 'Dirección de entrega', 'Sección de pago'];
const props = {
    activeStep,
    handleBack,
    handleNext,
    title: "Dirección de entrega",
    steps
}
const props2 = {
    activeStep: 2,
    handleBack,
    handleNext,
    title: "Dirección de entrega",
    steps
}


describe('Component DeliveryForm', () => {
    test("Component should render", () => {
    const { baseElement } =
    render(
        <ContextProvider>
            <DeliveryForm {...props} />
        </ContextProvider>
    )
    expect(baseElement).toMatchSnapshot();
    })

    test("Should show 3 labels and inputs", () => {
        render(
            <ContextProvider>
                <DeliveryForm {...props} />
            </ContextProvider>
        )
        const directionLabel = screen.getByLabelText('Dirección')
        const directionInput = screen.getByRole('textbox', {name: /dirección/i,})
        const departmentLabel = screen.getByLabelText('Departamento')
        const departmentInput = screen.getByRole('textbox', {name: /Departamento/i,})
        const citylLabel = screen.getByLabelText('Ciudad')
        const cityInput = screen.getByRole('textbox', {name: /Ciudad/i,})
        const stateLabel = screen.getByLabelText('Provincia')
        const stateInput = screen.getByRole('textbox', {name: /Provincia/i,})
        const zipCodeLabel = screen.getByLabelText('Codigo Postal')
        const zipCodeInput = screen.getByRole('textbox', {name: /Codigo Postal/i,})
        expect(directionLabel).toBeInTheDocument();
        expect(departmentLabel).toBeInTheDocument();
        expect(citylLabel).toBeInTheDocument();
        expect(stateLabel).toBeInTheDocument();
        expect(zipCodeLabel).toBeInTheDocument();
        expect(directionInput).toBeInTheDocument();
        expect(departmentInput).toBeInTheDocument();
        expect(cityInput).toBeInTheDocument();
        expect(stateInput).toBeInTheDocument();
        expect(zipCodeInput).toBeInTheDocument();
    })
    test('should render the submit button', () => {
        render(
            <ContextProvider>
                <DeliveryForm {...props} />
            </ContextProvider>
        )
        const submitButton = screen.getByRole('button', {name: /Proximo/i,})
        expect(submitButton).toBeInTheDocument();
    });
    test('should have focus on direction', () => {
        render(
            <ContextProvider>
                <DeliveryForm {...props} />
            </ContextProvider>
        )
        const directionInput = screen.getByRole('textbox', {name: /dirección/i,})
        expect(directionInput).toHaveFocus();
    });

    describe("Validating form", () => {
        test('should return true', async () => {
            const delivery: Delivery = {
                direccion: 'valid',
                departamento: 'valid',
                ciudad: 'valid',
                provincia: 'valid',
                codigoPostal: 'valid'
            }
            expect(await deliverySchema.isValid(delivery)).toBeTruthy();
        })
        test('should return false', async () => {
            const delivery: Delivery = {
                direccion: '',
                departamento: 'valid',
                ciudad: 'valid',
                provincia: 'valid',
                codigoPostal: 'valid'
            }
            expect(await deliverySchema.isValid(delivery)).toBeFalsy();
        })
    })

    describe("Submit Form", () => {
        test("Should dispatch info when submited", async () => {
            render(
                <ContextProvider>
                    <DeliveryForm {...props} />
                </ContextProvider>
            )
            
            const departmentnInput = screen.getByRole('textbox', {name: /departamento/i,})
            await userEvent.type(departmentnInput, "{backspace}{backspace}{backspace}{backspace}")
            await userEvent.type(departmentnInput, '14 F')
            const zipCodeInput = screen.getByRole('textbox', {name: /Codigo Postal/i,})
            await userEvent.type(zipCodeInput, "{backspace}{backspace}{backspace}{backspace}")
            await userEvent.type(zipCodeInput, '1234')

            const submitButton = screen.getByRole('button', {name: /Proximo/i,})
            userEvent.click(submitButton);

            await waitFor(() => {
                expect(handleNext).toBeCalled();
            })
            expect(mockDispatch).toBeCalledWith({
                payload: {
                    direccion: "Avengers Tower",
                    departamento: "14 F",
                    ciudad: "New York",
                    provincia: "New York",
                    codigoPostal: "1234",
                },
                type: "SUBMIT_DELIVERY"
            })
            })
    })

    describe("Last Step", () => {
        test("This is te last step", () => {
            render(
                <ContextProvider>
                    <DeliveryForm {...props2} />
                </ContextProvider>
            )

            const submitButton = screen.getByRole('button', {name: /Finalizar/i,})
            expect(submitButton).toBeInTheDocument();
        })
    })
})