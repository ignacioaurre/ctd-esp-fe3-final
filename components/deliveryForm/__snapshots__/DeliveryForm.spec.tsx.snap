// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component DeliveryForm Component should render 1`] = `
<body>
  <div>
    <form>
      <h4>
        Dirección de entrega
      </h4>
      <div
        class="css-nen11g-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="direccion"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r0:"
              id=":r0:-label"
            >
              Dirección 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby=":r0:-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="text"
                value="Avengers Tower"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Dirección 
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-focused MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              id=":r0:-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="departamento"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r1:"
              id=":r1:-label"
            >
              Departamento 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby=":r1:-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                type="text"
                value="20 A"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Departamento 
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              id=":r1:-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="ciudad"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r2:"
              id=":r2:-label"
            >
              Ciudad 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby=":r2:-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                type="text"
                value="New York"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Ciudad 
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              id=":r2:-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="provincia"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r3:"
              id=":r3:-label"
            >
              Provincia 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby=":r3:-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                type="text"
                value="New York"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Provincia 
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              id=":r3:-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="codigoPostal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r4:"
              id=":r4:-label"
            >
              Codigo Postal 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby=":r4:-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                type="text"
                value="9827"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Codigo Postal 
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
              id=":r4:-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-2royc9"
      >
        <button
          class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-a1sg1c-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Anterior
        </button>
        <div
          class="MuiBox-root css-11qjisw"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Proximo
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</body>
`;
